### PR TITLE
Update `docs/source/llms/deployments/index.rst`

### PR DESCRIPTION
**Pull request by harupy**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/harupy/mlflow/pull/10550?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/10550/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 10550
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

- Replace gateway with deployments
- Replace route(s) with endpoint(s)

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: docs/source/llms/deployments/index.rst [hunk 1]
```diff
@@ -693,25 +693,25 @@ For specifying an API key, there are three options:
...
+The MLflow Deployments Server defines standard parameters for chat, completions, and embeddings that can be
+used when querying any endpoint regardless of its provider. Each parameter has a standard range and
+default value. When querying an endpoint with a particular provider, the MLflow Deployments Server automatically
 scales parameter values according to the provider's value ranges for that parameter.
 
 Completions
 ~~~~~~~~~~~
 
-The standard parameters for completions routes with type ``llm/v1/completions`` are:
+The standard parameters for completions endpoints with type ``llm/v1/completions`` are:
```
Comment (harupy) [inline docs/source/llms/deployments/index.rst:714]: We might want to mention that we confirm to the OpenAI schema somewhere.

File: docs/source/llms/deployments/index.rst [hunk 2]
```diff
@@ -4,24 +4,31 @@
 MLflow Deployments Server (Experimental)
 ========================================
 
-The MLflow AI Gateway service is a powerful tool designed to streamline the usage and management of
+.. important::
+   The feature previously known as **MLflow AI Gateway** in experimental status has been moved to
+   utilize the **MLflow deployments API**. This major update involves changes to API endpoints and
+   standardization for Large Language Models, both custom and SaaS-based. Users currently utilizing
+   MLflow AI Gateway should refer to the new documentation for migration guidelines and familiarize
+   themselves with the updated API structure.
```
Comment (harupy) [inline docs/source/llms/deployments/index.rst:12]: @BenWilson2 for addressing https://github.com/mlflow/mlflow/pull/10522#discussion_r1410108654

File: docs/source/llms/deployments/index.rst [hunk 3]
```diff
@@ -33,35 +40,35 @@ organizations that use LLMs on a regular basis.
...
 
 .. code-block:: sh
 
-    pip install 'mlflow[gateway]'
+    pip install 'mlflow[genai]'
 
 Step 2: Set the OpenAI API Key(s) for each provider
 ---------------------------------------------------
-The Gateway service needs to communicate with the OpenAI API. To do this, it requires an API key.
+The deployments server needs to communicate with the OpenAI API. To do this, it requires an API key.
```
Comment (harupy) [inline docs/source/llms/deployments/index.rst:71]: Let me know if there are `server` that should be capitalized/uncapitalized.

File: docs/source/llms/deployments/index.rst [hunk 4]
```diff
@@ -176,46 +183,46 @@ Firstly, update the :ref:`MLflow AI Gateway config <deployments_configuration>`
           config:
             openai_api_key: $OPENAI_API_KEY
 
-This updated configuration adds a new completions route ``completions-gpt4`` while still preserving the original ``completions``
-route that was configured with the ``gpt-3.5-turbo``  model.
+This updated configuration adds a new completions endpoint ``completions-gpt4`` while still preserving the original ``completions``
+endpoint that was configured with the ``gpt-3.5-turbo``  model.
 
-Once the configuration file is updated, simply save your changes. The Gateway will automatically create the new route with zero downtime.
+Once the configuration file is updated, simply save your changes. The Deployments Server will automatically create the new endpoint with zero downtime.
```
Comment (BenWilson2) [inline docs/source/llms/deployments/index.rst:189]: this one is a generic reference and can be lowercased

```suggestion
Once the configuration file is updated, simply save your changes. The deployments server will automatically create the new endpoint with zero downtime.
```

File: docs/source/llms/deployments/index.rst [hunk 5]
```diff
@@ -111,23 +118,23 @@ For details about the configuration file's parameters (including parameters for
...
 
-Save this file to a location on the system that is going to be running the MLflow AI Gateway server.
+Save this file to a location on the system that is going to be running the MLflow Deployments Server.
 
-Step 4: Start the Gateway Service
----------------------------------
-You're now ready to start the Gateway service!
+Step 4: Start the Deployments Server
+------------------------------------
+You're now ready to start the Deployments Server!
```
Comment (BenWilson2) [inline docs/source/llms/deployments/index.rst:125]:
```suggestion
You're now ready to start the deployments server!
```

File: docs/source/llms/deployments/index.rst [hunk 6]
```diff
@@ -176,46 +183,46 @@ Firstly, update the :ref:`MLflow AI Gateway config <deployments_configuration>`
...
-This updated configuration adds a new completions route ``completions-gpt4`` while still preserving the original ``completions``
-route that was configured with the ``gpt-3.5-turbo``  model.
+This updated configuration adds a new completions endpoint ``completions-gpt4`` while still preserving the original ``completions``
+endpoint that was configured with the ``gpt-3.5-turbo``  model.
 
-Once the configuration file is updated, simply save your changes. The Gateway will automatically create the new route with zero downtime.
+Once the configuration file is updated, simply save your changes. The Deployments Server will automatically create the new endpoint with zero downtime.
 
-If you no longer need a route, you can delete it from the configuration YAML and save your changes. The AI Gateway will automatically remove the route.
+If you no longer need an endpoint, you can delete it from the configuration YAML and save your changes. The Deployments Server will automatically remove the endpoint.
```
Comment (BenWilson2) [inline docs/source/llms/deployments/index.rst:191]:
```suggestion
If you no longer need an endpoint, you can delete it from the configuration YAML and save your changes. The deployments server will automatically remove the endpoint.
```

File: docs/source/llms/deployments/index.rst [hunk 7]
```diff
@@ -176,46 +183,46 @@ Firstly, update the :ref:`MLflow AI Gateway config <deployments_configuration>`
...
 
-Step 9: Use AI Gateway routes for model development
-----------------------------------------------------
+Step 9: Use Deployments Server endpoints for model development
+--------------------------------------------------------------
 
-Now that you have created several AI Gateway routes, you can create MLflow Models that query these
-routes to build application-specific logic using techniques like prompt engineering. For more
-information, see :ref:`AI Gateway and MLflow Models <deployments_mlflow_models>`.
+Now that you have created several Deployments Server endpoints, you can create MLflow Models that query these
```
Comment (BenWilson2) [inline docs/source/llms/deployments/index.rst:196]:
```suggestion
Now that you have created several deployments server endpoints, you can create MLflow Models that query these
```

File: docs/source/llms/deployments/index.rst [hunk 8]
```diff
@@ -291,33 +298,33 @@ As of now, the MLflow AI Gateway supports the following providers:
...
 * **bedrock**: This is used for models offered by `AWS Bedrock <https://aws.amazon.com/bedrock/>`_.
 
-More providers are being added continually. Check the latest version of the MLflow AI Gateway Docs for the
+More providers are being added continually. Check the latest version of the MLflow Deployments Server Docs for the
 most up-to-date list of supported providers.
 
-Remember, the provider you specify must be one that the MLflow AI Gateway supports. If the provider
-is not supported, the Gateway will return an error when trying to route requests to that provider.
+Remember, the provider you specify must be one that the MLflow Deployments Server supports. If the provider
+is not supported, the Deployments Server will return an error when trying to route requests to that provider.
```
Comment (BenWilson2) [inline docs/source/llms/deployments/index.rst:305]:
```suggestion
is not supported, the deployments server will return an error when trying to route requests to that provider.
```

File: docs/source/llms/deployments/index.rst (hunk 1 above)
Comment (BenWilson2) [reply docs/source/llms/deployments/index.rst:721]: Could be a nice section to add under `Querying the Deployments Server` 

``` rst
API Syntax Standards
----------------------

The MLflow Deployments Server endpoint interfaces conform to those of the standards established by OpenAI. Other providers input and output interfaces will be translated to the OpenAI specification in order to provide a single standardized API interface when using the MLflow Deployments Server. 
```

WDYT?

File: docs/source/llms/deployments/index.rst [hunk 9]
```diff
@@ -931,26 +942,26 @@ This integration enable users to use prompt engineering, retrieval augmented gen
 
 MLflow Models
 ~~~~~~~~~~~~~
-Interfacing with MLflow Models can be done in two ways. With the use of a custom PyFunc Model, a query can be issued directly to an AI Gateway endpoint and used in a broader context within a model.
-Data may be augmented, manipulated, or used in a mixture of experts paradigm. The other means of utilizing the AI Gateway along with MLflow Models is to define a served MLflow model directly as a
-route within the AI Gateway.
+Interfacing with MLflow Models can be done in two ways. With the use of a custom PyFunc Model, a query can be issued directly to an Deployments Server endpoint and used in a broader context within a model.
```
Comment (BenWilson2) [inline docs/source/llms/deployments/index.rst:945]:
```suggestion
Interfacing with MLflow Models can be done in two ways. With the use of a custom PyFunc Model, a query can be issued directly to a deployments server endpoint and used in a broader context within a model.
```

File: docs/source/llms/deployments/index.rst [hunk 10]
```diff
@@ -931,26 +942,26 @@ This integration enable users to use prompt engineering, retrieval augmented gen
...
+Data may be augmented, manipulated, or used in a mixture of experts paradigm. The other means of utilizing the Deployments Server along with MLflow Models is to define a served MLflow model directly as
```
Comment (BenWilson2) [inline docs/source/llms/deployments/index.rst:946]:
```suggestion
Data may be augmented, manipulated, or used in a mixture of experts paradigm. The other means of utilizing the MLflow Deployments Server along with MLflow Models is to define a served MLflow model directly as
```

File: docs/source/llms/deployments/index.rst [hunk 11]
```diff
@@ -931,26 +942,26 @@ This integration enable users to use prompt engineering, retrieval augmented gen
...
-You can also build and deploy MLflow Models that call the MLflow AI Gateway.
-The example below demonstrates how to use an AI Gateway server from within a custom ``pyfunc`` model.
+You can also build and deploy MLflow Models that call the MLflow Deployments Server.
+The example below demonstrates how to use an Deployments Server from within a custom ``pyfunc`` model.
 
 
 .. note::
-    The custom ``Model`` shown in the example below is utilizing environment variables for the AI Gateway server's uri. These values can also be set manually within the
-    definition or can be applied via :func:`mlflow.gateway.get_gateway_uri` after the uri has been set. For the example below, the value for ``MLFLOW_GATEWAY_URI`` is
+    The custom ``Model`` shown in the example below is utilizing environment variables for the Deployments Server's uri. These values can also be set manually within the
```
Comment (BenWilson2) [inline docs/source/llms/deployments/index.rst:963]:
```suggestion
    The custom ``Model`` shown in the example below is utilizing environment variables for the deployments server's uri. These values can also be set manually within the
```

File: docs/source/llms/deployments/index.rst [hunk 12]
```diff
@@ -1006,9 +1017,9 @@ This custom MLflow model can be used in the same way as any other MLflow model.
 
 REST API
 ~~~~~~~~
-The REST API allows you to send HTTP requests directly to the MLflow AI Gateway server. This is useful if you're not using Python or if you prefer to interact with the Gateway using HTTP directly.
+The REST API allows you to send HTTP requests directly to the MLflow Deployments Server. This is useful if you're not using Python or if you prefer to interact with the Deployments Server using HTTP directly.
```
Comment (BenWilson2) [inline docs/source/llms/deployments/index.rst:1020]:
```suggestion
The REST API allows you to send HTTP requests directly to the MLflow Deployments Server. This is useful if you're not using Python or if you prefer to interact with a deployments server using HTTP directly.
```

File: docs/source/llms/deployments/index.rst [hunk 13]
```diff
@@ -1006,9 +1017,9 @@ This custom MLflow model can be used in the same way as any other MLflow model.
...
 
-Here are some examples for how you might use curl to interact with the Gateway:
+Here are some examples for how you might use curl to interact with the Deployments Server:
```
Comment (BenWilson2) [inline docs/source/llms/deployments/index.rst:1022]:
```suggestion
Here are some examples for how you might use curl to interact with the MLflow Deployments Server:
```

**Review by BenWilson2 — APPROVED** [review]
Handful of consistency nits for referring to the proper noun form vs. the generic reference to the service

**Comment by harupy** [comment]
@BenWilson2 Thanks for the suggestion, batch-commited them.

File: docs/source/llms/deployments/index.rst (hunk 1 above)
Comment (harupy) [reply docs/source/llms/deployments/index.rst:721]: Sounds good. I'll add this in a separate PR.

**Pull request merged**